x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/FieldHitExtractor.java

```diff
@@ -143,8 +148,12 @@ private Object unwrapMultiValue(Object values) {
         }
         if (dataType == DataType.GEO_SHAPE) {
             Object value;
-            if (values instanceof List && ((List<?>) values).size() == 1) {
-                value = ((List<?>) values).get(0);
+            if (values instanceof List) {
```

Comment:
Could you please extract it to a method as it seems duplicate with the `GEO_POINT` method?